DOC: Validating that the word pandas is correctly capitalized

joybh98 opened the pull request:
- [x] closes #32316 
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

As suggested by @datapythonista , I've added checks in `ci/code_checks.sh` to look if the `pandas` is being referenced in a standardized way i.e `pandas`,  not ` *pandas* `  or  `Pandas`

@datapythonista I've only added checks to see if pandas is being referenced correctly, but not showing where it's changed or how many references are wrong. 
Should I add checks for that also ? What do you think ?

datapythonista commented:
Can you have a look at the diff? I think something went wrong, there are lots of unrelated changes.

joybhallaa commented:
@datapythonista you're right, let me check what went wrong.

MomIsBestFriend commented:
@joybhallaa Can you run the following and push?

```
git checkout add-check-standard-pandas
git fetch upstream
git merge upstream/master
git reset --soft upstream/master
git commit -m "Fixed wrong diff"
```

---

Taken all of the commands above from https://datapythonista.me/blog/useful-git-commands.html

joybhallaa commented:
@MomIsBestFriend okay, I'll do that !

joybhallaa commented:
> ```
> git checkout add-check-standard-pandas
> git fetch upstream
> git merge upstream/master
> git reset --soft upstream/master
> git commit -m "Fixed wrong diff"
> ```

@MomIsBestFriend I executed all these commands, and when pushing using `git push origin add-check-standard-pandas`, I got the error `hint: Updates were rejected because the tip of your current branch is behind
hint: its remote counterpart. Integrate the remote changes (e.g.
hint: 'git pull ...') before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details.`

Let me check what's causing the problem and I'll push ASAP

MomIsBestFriend commented:
@joybhallaa Have you followed the steps in the [Contributing guide](https://dev.pandas.io/docs/development/contributing.html#id9)?

datapythonista commented:
@joybhallaa first try `git diff upstream/master` and see if now you've got the changes you're proposing in this PR. If that's the case, you need a `git push --force`, since you're rewriting the history here (replacing all these unwanted commits by the new single commit you just did).

joybh98 committed:
Fixed wrong diff

joybhallaa commented:
> @joybhallaa first try `git diff upstream/master` and see if now you've got the changes you're proposing in this PR. If that's the case, you need a `git push --force`, since you're rewriting the history here (replacing all these unwanted commits by the new single commit you just did).

@datapythonista done

datapythonista commented on ci/code_checks.sh:
```diff
@@ -80,6 +80,13 @@ if [[ -z "$CHECK" || "$CHECK" == "lint" ]]; then
+    # Check if pandas is referenced as pandas, not *pandas* or Pandas
+    MSG='Checking if pandas reference is standardized or not' ; echo  $MSG
+    grep -nr '*pandas*|Pandas' doc/*
+    grep -nr '*pandas*|Pandas' web/*
```
I guess the next code works. You can't have two statements, since you are checking the exit code later, and you're only checking the code of the last statement.
```suggestion
    grep -nr '*pandas*|Pandas' web/* doc/*
```
Also, please use `invgrep` as in the rest of this script, you're doing here the opposite of what you want to do. And we have lots of validations where we already do this for other patterns, don't need to reinvent the wheel, just so the same, so we also keep consistent.

datapythonista commented on ci/code_checks.sh:
```diff
@@ -80,6 +80,13 @@ if [[ -z "$CHECK" || "$CHECK" == "lint" ]]; then
+    grep -nr '*pandas*|Pandas' web/*
+    RET=$(($RET + $?)) ; echo $MSG "DONE"
+
+
```
Don't leave extra blank lines, keep the same style and formatting as the rest of the script.

datapythonista reviewed (changes requested):
You are adding this code in the linting section (flake8...). We have a section named patterns where we are checking for many patterns in the same way as you need to check for this one.

datapythonista commented:
@joybhallaa do you have time to have a look at the previous comments?

joybhallaa commented:
@datapythonista yep was busy with college stuff. Will start work ASAP

joybh98 committed:
added requested changes

joybhallaa commented on ci/code_checks.sh:
```diff
@@ -229,6 +229,11 @@ if [[ -z "$CHECK" || "$CHECK" == "patterns" ]]; then
     unset INVGREP_APPEND
+
+    # Check if pandas is referenced as pandas, not *pandas* or Pandas
+    MSG='Checking if the pandas word reference is always used lowercase (pandas,not Pandas or PANDAS' ; echo  $MSG
```
@datapythonista I forgot to add `)` on line 234, please let me know if everything else besides this right or not.

datapythonista reviewed (commented):
CI is green, and I don't think we fixed all the cases of `Pandas` and `*pandas*` in the docs and web, did we? If we didn't, this is not working.

We use `invgrep -R` everywhere else, I think we should use the same, may be that's the problem.

Also, can you add `PANDAS` please?

MomIsBestFriend commented on ci/code_checks.sh:
```diff
@@ -229,6 +229,11 @@ if [[ -z "$CHECK" || "$CHECK" == "patterns" ]]; then
+
+    # Check if pandas is referenced as pandas, not *pandas* or Pandas
+    MSG='Checking if the pandas word reference is always used lowercase (pandas,not Pandas or PANDAS' ; echo  $MSG
+    invgrep -nr '*pandas*|Pandas' web/* doc/*
```
I think the problem is with the astriks in the end, does the CI fail if we remove those?

joybhallaa replied:
Let me check

joybhallaa replied:
@MomIsBestFriend CI passes even after removing the asterisk

joybhallaa commented:
> CI is green, and I don't think we fixed all the cases of `Pandas` and `*pandas*` in the docs and web, did we? If we didn't, this is not working.
> 
> We use `invgrep -R` everywhere else, I think we should use the same, may be that's the problem.
> 
> Also, can you add `PANDAS` please?

Will do

joybh98 committed:
added -R and PANDAS

ShaharNaveh committed:
CI: add doctest check for done modules (#33175)

datapythonista commented:
Something is wrong here, @joybhallaa can you have a look at the diff, and leave things so this is ready for review please?

jbrockmendel committed:
REF: .get tests (#33164)

ShaharNaveh committed:
CLN: using C-API of datetime (#33174)

* CLN: cimporting time instead of going to python space

pandas/_libs/tslibs/conversion.pyx

* cimporting time instead of going to python space

pandas/_libs/tslibs/timestamps.pyx

* cimporting time instead of going to python space

pandas/_libs/tslibs/period.pyx

Co-authored-by: MomIsBestFriend <>

Commit:
DOC: Partial fix SA04 errors in docstrings #28792 (feedback needed) (#32823)

jbrockmendel committed:
REF: fillna tests (#33183)

WillAyd committed:
Json parametrize more2 (#33163)

jbrockmendel committed:
CLN: use ._data less in reshape (#33159)

jbrockmendel committed:
BUG: isna_old with td64, dt64tz, period (#33158)

jbrockmendel committed:
REF: .drop tests (#33156)

jbrockmendel committed:
REF: test_reindex_like (#33150)

topper-123 committed:
BUG: create new MI from MultiIndex._get_level_values (#33134)

mroeschke committed:
BUG: to_datetime with infer_datetime_format dropped timezone names (#33133)

themien committed:
DOC: Fix capitalization of doc section headers (#33149)

themien committed:
DOC: Fix capitalization of docs section headers (#32944)

farhanreynaldo committed:
DOC: Fix PR06 (type names) in docstrings (#33093)

Co-authored-by: Farhan Reynaldo <farhanreynaldo@gmail.com>

jamescobonkerr committed:
TST: cover search_sorted scalar mixed timezones case (#33185)

MarianD committed:
DOC: Added docstring for show_versions() (#33073)

jbrockmendel committed:
REF: set_axis tests (#33189)

SultanOrazbayev committed:
ENH/VIZ: Allowing `s` parameter of scatter plots to be a column name (#33107)

jbrockmendel committed:
misplaced sort-index test (#33191)

tv3141 committed:
Correct docstring to describe actual functionality. (#33190)

jbrockmendel committed:
REF/CLN: test_get_dummies (#33184)

jbrockmendel committed:
TYP: require _update_inplace gets Frame/Series, never BlockManager (#33074)

* TYP: require _update_inplace gets BlockManager

* standardize on requiring Series/DataFrame, never BlockManager

joybhallaa commented:
@datapythonista yes.Working on it right now

joybh98 committed:
fixed merge conflicts

joybhallaa commented:
@datapythonista should I open a new PR with a new branch ? This branch keeps giving me problems.

MomIsBestFriend commented:
@joybhallaa I suggest you read https://datapythonista.me/blog/useful-git-commands.html

> I've got some cool changes, but my history is a mess

joybh98 committed:
All changes in a single commit

joybh98 committed:
fixed merge conflict in test_unstack.py

joybhallaa commented:
@datapythonista the 2nd last commit only has the changes that I've made. 
Let me know if that's suitable

datapythonista commented:
> @datapythonista the 2nd last commit only has the changes that I've made.
> Let me know if that's suitable

No, we need a proper PR to be able to fix. @MomIsBestFriend gave you a resource to fix this. But probably better to open a new PR, closing this one, since your changes are small, and looks like it'll be easier to start again, than to fix this.
